lib/key.rb

```diff
@@ -1,17 +1,32 @@
 class Key
-  attr_reader :number
+  attr_reader :number, :key_hash
```

Comment:
do i need to keep the :key_hash as an attribute reader or should it be a method?